### DISCUSSION TITLE
What should I do to switch the theme of uno platform?

### DISCUSSION THREAD

**Discussion by CeSun** [unoplatform/uno#10176] [Q&A] [answered]
I don't understand the documentation, which package should I add to which project and which package should I remove?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the new discussion. Which documentation are you referring to?

**Reply by CeSun** [reply]
Thank you for your replay.

https://platform.uno/docs/articles/external/uno.themes/doc/cupertino-getting-started.html

> Choose [Uno.Cupertino](https://www.nuget.org/packages/Uno.Cupertino/) package for Uno UWP solution or choose the [Uno.Cupertino.WinUI](https://www.nuget.org/packages/Uno.Cupertino.WinUI) package for Uno WinUI solution.

Whitch project is UWP project and whitch one is WinUI solution?

![image](https://user-images.githubusercontent.com/54771283/196469995-0f625ba9-fa0a-42b8-855c-0934cc45e27b.png)

**Reply by jeromelaban** [reply]
Indeed it's not clear. Your solution uses WinUI, you can find it by looking at your project nuget package references (`Uno.WinUI` for `WinUI`, `Uno.UI` for `UWP`).

You can find [an example of theme switching here](https://github.com/unoplatform/Uno.Gallery/blob/7677ac022e072e04ecaba492a4dc1b9fb020b737/Uno.Gallery/Uno.Gallery.UWP/Views/Shell.xaml.cs#L90-L115).

**Reply by CeSun** [reply]
> Indeed it's not clear. Your solution uses WinUI, you can find it by looking at your project nuget package references ( for , for ).`Uno.WinUI``WinUI``Uno.UI``UWP`
> 
> You can find [an example of theme switching here](https://github.com/unoplatform/Uno.Gallery/blob/7677ac022e072e04ecaba492a4dc1b9fb020b737/Uno.Gallery/Uno.Gallery.UWP/Views/Shell.xaml.cs#L90-L115).

I find `Uno.WinUI` reference on my gtksharp project, it means it's a WinUI project?

**Reply by CeSun** [reply]
> Thanks for the new discussion. Which documentation are you referring to?

According to your guide, I have solved the problem, thanks again for your reply!

**Comment by acrigney** [comment]
Graet but what about if I want to build and use a custom theme? How can I do that.

**Reply by jeromelaban** [reply]
WinUI only provides a few non-configurable (by name only: Light, Dark, High-Contrast) themes. Yet, you can alter anything in the UI by using custom styles. Why do you need to change?

Also to be sure, what do you call a theme? That word has a very specific meaning in WinUI that may not be what you attribute it to.

**Comment by acrigney** [comment]
I thought I would be able to build a corporate theme like a boostrap
template or even pay experts to build that for us.
